Tasks/DeployAzureResourceGroup/MGExtensionManager.ts

```diff
@@ -0,0 +1,132 @@
+var computeManagementClient = require("azure-arm-compute");
+import util = require("util");
+import tl = require("vsts-task-lib/task");
+import azure_utils = require("./AzureUtil");
+import deployAzureRG = require("./DeployAzureRG");
+import Q = require('q');
+
+export class MGExtensionManager {
+    private taskParameters: deployAzureRG.AzureRGTaskParameters;
+    private credentials;
+    private subscriptionId: string;
+    private azureUtils;
+    private failureCount: number;
+    private successCount: number;
+    private errors: string;
+    private vmCount: number;
+    private deferred;
+    private operation: string;
+    private computeClient;
+
+    constructor(taskParameters: deployAzureRG.AzureRGTaskParameters) {
+        this.taskParameters = taskParameters;
+        this.credentials = this.taskParameters.credentials;
+        this.subscriptionId = this.taskParameters.subscriptionId;
+        this.azureUtils = new azure_utils.AzureUtil(this.taskParameters);
+        this.successCount = 0;
+        this.failureCount = 0;
+        this.vmCount = 0;
+        this.operation = "";
+
+        this.computeClient = new computeManagementClient(this.credentials, this.subscriptionId);
+        return this;
+    }
+
+    private setTaskResult() {
+        if (this.failureCount + this.successCount == this.vmCount) {
+            if (this.failureCount > 0) {
+                console.log("Machine group agent " + this.operation + " did not succeed on all VMs");
+            }
+            else {
+                console.log("Machine group agent " + this.operation + " succeeded on all VMs");
+            }
+            this.deferred.resolve("");
+        }
+    }
+
+    private postOperationCallBack = (error, result, request, response) => {
+        if (error) {
+            this.failureCount++;
+            this.errors += error.message;
+            this.errors += "\n";
+        } else {
+            this.successCount++;
+        }
+        this.setTaskResult();
+    }
+
+    public async installMGExtension() {
+        this.deferred = Q.defer<string>();
+        this.operation = "installation";
+        var listOfVms = await this.azureUtils.getVMDetails();
+        this.vmCount = listOfVms.length;
+        if(this.vmCount == 0){
+            this.deferred.resolve("");
+        }
+        for (var i = 0; i < listOfVms.length; i++) {
+            var vmName = listOfVms[i]["name"];
+            var extensionParameters = this.FormExtensionParameters(listOfVms[i], "enable");
+            console.log("Adding " + extensionParameters["extensionName"] + " extension to virtual machine " + vmName);
```

Comment:
Add these logs + some additional logs like 'public setting content' in tl.debug.
For each VM just print the extension name being enabled.